librandom/gsl_binomial_randomdev.cpp

```diff
@@ -84,13 +84,27 @@ librandom::GSL_BinomialRandomDev::set_p_n( double p_s, unsigned n_s )
 void
 librandom::GSL_BinomialRandomDev::set_p( double p_s )
 {
-  assert( 0.0 <= p_ && p_ <= 1.0 );
+  if ( p_s < 0. or 1. < p_s )
+  {
+    throw BadParameterValue( "gsl_binomial RDV: 0 <= p <= 1 required." );
+  }
   p_ = p_s;
 }
 
 void
-librandom::GSL_BinomialRandomDev::set_n( unsigned int n_s )
+librandom::GSL_BinomialRandomDev::set_n( size_t n_s )
 {
+  // gsl_ran_binomial() takes n as an unsigned int, so it cannot be
+  // greater than what an unsigned int can hold.
+  const auto N_MAX = std::numeric_limits< unsigned int >::max();
+  if ( n_s >= N_MAX )
+  {
+    throw BadParameterValue( String::compose( "Gsl_binomial RDV: N < %1 required.", static_cast< double >( N_MAX ) ) );
```

Comment:
I'd suggest lowercase `gsl_binomial` in error message.

Reply:
@heplesser Done.